tensorboard/webapp/header/dark_mode_toggle_component.ts

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+
+export enum DarkModeOverride {
+  DEFAULT,
+  DARK_MODE_ON,
+  DARK_MODE_OFF,
+}
+
+@Component({
+  selector: 'app-header-dark-mode-toggle-component',
+  template: `
+    <button
+      mat-icon-button
+      [matMenuTriggerFor]="menu"
+      aria-label="Menu for changing light or dark theme"
+      [ngSwitch]="darkModeOverride"
+      title="Switch between browser default, light, or dark theme"
+    >
+      <mat-icon
+        *ngSwitchCase="DarkModeOverride.DEFAULT"
+        svgIcon="brightness_6_24px"
+      ></mat-icon>
+      <mat-icon
+        *ngSwitchCase="DarkModeOverride.DARK_MODE_OFF"
+        svgIcon="light_mode_24px"
+      ></mat-icon>
+      <mat-icon
+        *ngSwitchCase="DarkModeOverride.DARK_MODE_ON"
+        svgIcon="dark_mode_24px"
+      ></mat-icon>
+    </button>
+    <mat-menu #menu="matMenu">
+      <button
+        mat-menu-item
+        (click)="onOverrideChanged.emit(DarkModeOverride.DEFAULT)"
+      >
+        <label>Browser default</label>
```

Comment:
Is the text `Default` or `Browser default`? (I'm seeing the former in the screenshot)

Reply:
It is latter now. I had changed it after taking the screenshot to make it more verbose and correct.